src/graylint/output/destination.py

```diff
@@ -0,0 +1,64 @@
+"""Encapsulate path/device/stdout handling in a class."""
+
+from __future__ import annotations
+
+import sys
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import TextIO
+
+
+class OutputDestination:
+    """Encapsulate path/device/stdout handling in a class."""
+
+    def __init__(self, path_or_dash: Path | str) -> None:
+        """Initialize the output destination class."""
+        if isinstance(path_or_dash, str) and path_or_dash == "-":
+            self._path: Path | None = None
```

Comment:
Maybe declare the property types in the class declaration inside of inside the constructor?

Reply:
I can see three ways to define the types in the class declaration:
1. Include the types in the class docstring.
2. Define `self._path` and `self._file` as class attributes.
3. Turn the class into a dataclass.

Option 1. is quite outdated, since type annotations are now supported in Python.

Option 2. is wrong in my opinion, since these are instance attributes. There's no reason to use class attributes here.

Option 3. is a misuse of dataclasses, I think. This is more than a simple data storage class.

Is there a reason you think it's not a good practice to add typing information inside methods, or the constructor method in particular?